l10n_es_account_asset_variation/models/account_asset.py

```diff
@@ -0,0 +1,167 @@
+                                  digits=dp.get_precision('Account'))
+    sequence = fields.Char(string='Code', default='/')
+
+    @api.multi
+    def _get_last_depreciation_date(self):
+        res = {}
+        for record in self:
+            done_lines = record.depreciation_line_ids.filtered(lambda x:
+                                                               x.move_id)
+            date = max([record.purchase_date] +
+                       done_lines.mapped('depreciation_date'))
+            max_date = fields.Date.from_string(date)
+            if done_lines and record.prorata:
+                max_date = (max_date +
+                            relativedelta(months=+record.method_period))
+            res.update({record.id: fields.Date.to_string(max_date)})
+        return res
+
+    @api.multi
+    def _get_real_depreciation_date(self, depr_date):
+        self.ensure_one()
+        if self.method_period == 12:
+            depr_date = depr_date.replace(depr_date.year, 12, 31)
+        return fields.Date.to_string(depr_date)
+
+    @api.model
+    def _compute_board_amount(
+            self, asset, i, residual_amount, amount_to_depr,
+            undone_dotation_number, posted_depreciation_line_ids, total_days,
+            depreciation_date):
+        wiz = self.env.context.get('wiz', False)
+        dep_date = asset._get_real_depreciation_date(depreciation_date)
+        if not wiz or asset.method_time != 'percentage':
+            return super(AccountAssetAsset, self)._compute_board_amount(
+                asset, i, residual_amount, amount_to_depr,
+                undone_dotation_number, posted_depreciation_line_ids,
+                total_days, depreciation_date)
+        amount = 0
+        if i == undone_dotation_number:
+            amount = residual_amount
+        else:
+            percentage2apply = asset.method_percentage
+            if dep_date >= wiz.start_date and dep_date <= wiz.end_date:
+                percentage2apply = wiz.percentage
+            if i == 1 and asset.prorata:
+                if asset.method_period == 1:
+                    total_days = calendar.monthrange(
+                        depreciation_date.year, depreciation_date.month)[1]
+                    days = total_days - float(depreciation_date.day) + 1
+                else:
+                    days = (total_days - float(
+                        depreciation_date.strftime('%j'))) + 1
+                percentage = percentage2apply * days / total_days
+            else:
+                percentage = percentage2apply
+            amount = asset.purchase_value * percentage / 100
+        return amount
+
+    @api.model
+    def _compute_board_undone_dotation_nb(self, asset, depreciation_date,
+                                          total_days):
+        wizard = self.env.context.get('wiz', False)
+        depreciation_date = fields.Date.from_string(asset.purchase_date)
+        if (asset.method_time == 'percentage' and wizard):
```

Comment:
Yo solo pondría una if:
if asset.method_time != 'percentage' or not wizard:
   return super(AccountAssetAsset, self).\
          _compute_board_undone_dotation_nb(
          asset, depreciation_date, total_days)
number = 0
percentage = 100.0
while percentage > 0:
      percentage2apply = asset.method_percentage
      dep_date = fields.Date.to_string(depreciation_date)
      if dep_date >= wizard.start_date and \
          xxxxxxxxxx
return number